### PR TITLE
III-3730 Apply top level status on sub events with no status

### PR DESCRIPTION
**Pull request by LucWollants**
### Changed
 
- Apply top-level status on sub-events with no status to avoid default available status
 
---
Ticket: https://jira.uitdatabank.be/browse/III-3770

### CODE REVIEW

**Review by jdrieghe — APPROVED** [review]
Looks good 👍

File: tests/Serializer/Event/EventDenormalizerTest.php [hunk 1]
```diff
@@ -669,6 +669,81 @@ public function it_should_denormalize_event_data_with_a_multiple_date_ranges_cal
         $this->assertEquals($expected, $actual);
     }
 
+    /**
+     * @test
+     */
+    public function it_should_denormalize_event_data_with_single_date_range_and_apply_status_to_sub_event_with_no_status()
```
Comment (bertramakers) [inline tests/Serializer/Event/EventDenormalizerTest.php:675]: Nice, like the TDD approach! 👍  I also try to do this, especially when making changes to existing functionality, but should do it even more often.

File: tests/Serializer/Event/EventDenormalizerTest.php [hunk 2]
```diff
@@ -669,6 +669,81 @@ public function it_should_denormalize_event_data_with_a_multiple_date_ranges_cal
...
+                new Category(
+                    new CategoryID('0.50.1.0.1')
+                )
+            )
+        );
+
+        $actual = $this->denormalizer->denormalize($eventData, ImmutableEvent::class);
+
+        $this->assertEquals($expected, $actual);
+    }
```
Comment (bertramakers) [inline tests/Serializer/Event/EventDenormalizerTest.php:746]: Nit: The test method sticks to the next test

Comment (bertramakers) [reply tests/Serializer/Event/EventDenormalizerTest.php:746]: Nevermind, fixed in a later commit 👍

**Review by bertramakers — APPROVED** [review]
Elegant solution! 👍

**Pull request merged**